homeassistant/components/waterkotte_heatpump/__init__.py

```diff
@@ -0,0 +1,59 @@
+"""The Waterkotte Heatpump integration."""
+from __future__ import annotations
+
+import logging
+
+from pywaterkotte import AuthenticationException, ConnectionException, Ecotouch
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, DOMAIN
+from .coordinator import EcotouchCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Waterkotte Heatpump from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
This should happen after authentication.

you can make it
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
on line 47